src/mame/mattel/pixtermu.cpp

```diff
@@ -0,0 +1,223 @@
+    - https://www.nxp.com/docs/en/user-guide/LH79524-LH79525_UG_V1_3.pdf
+    - https://lh79525.wordpress.com/
+
+    Hardware
+    --------
+
+    Model H4651/J4287/J4288:
+
+    - PCB Revision: PT1543A-BGA-4F2C 2005/04/30 Rev 4.2b
+    - CPU: Sharp LH79524-NOE (ARM720T)
+    - RAM: ICSI IC42S16100-7T (2MB each, 4MB total)
+    - ROM: Chip-On-Board, selected by pin CS1
+
+    JTAG
+    ----
+
+    To place the LH79524 into ARM ICE debug mode you will need to connect TEST1 via a 4.7K Ohm resistor to ground.
+
+    Pinout:
+
+    - D2: nTRST (Reset Input)
+    - P4: TMS (Mode Select Input)
+    - T3: TCK (Clock Input)
+    - T1: TDI (Serial Data Input)
+    - P3: TDO (Data Serial Output)
+    - TEST1: ARM ICE debug mode
+
+*******************************************************************************/
+
+#include "emu.h"
+
+#include "bus/generic/carts.h"
+#include "bus/generic/slot.h"
+#include "cpu/arm7/arm7.h"
+#include "cpu/arm7/arm7core.h"
+
+#include "softlist_dev.h"
+
+namespace {
+
+class pixter_multimedia_state : public driver_device
+{
+public:
+	pixter_multimedia_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_cart(*this, "cartslot")
+		, m_maincpu(*this, "maincpu")
+		, m_ndcs0(*this, "ndcs0")
+		, m_internal_sram(*this, "internal_sram")
+		, m_apb(*this, "apb", 0x27000, ENDIANNESS_LITTLE)
+		, m_remap_view(*this, "remap")
+	{ }
+
+	void pixter_multimedia(machine_config &config);
+
+private:
+	// Remap Control, mapped at 0xfffe2008, offset 0x22008/4
+	static inline constexpr uint32_t APB_REMAP = 34818;
+	// Power-up Boot Configuration, mapped at 0xfffe6000, offset 0x26000/4
+	static inline constexpr uint32_t APB_PBC = 38912;
+	// nCS1 Override, mapped at 0xfffe6004, offset 0x26004/4
+	static inline constexpr uint32_t APB_CS1OV = 38913;
+	// External Peripheral Mapping, mapped at 0xfffe6008, offset 0x26008/4
+	static inline constexpr uint32_t APB_EPM = 38914;
+
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+	DEVICE_IMAGE_LOAD_MEMBER(cart_load);
+	void arm7_map(address_map &map);
+	void apb_bridge_w(offs_t offset, uint32_t data, uint32_t mem_mask);
+	void apb_remap(uint32_t data);
+
+	required_device<generic_slot_device> m_cart;
+	required_device<arm7_cpu_device> m_maincpu;
+	required_shared_ptr<uint32_t> m_ndcs0;
+	required_shared_ptr<uint32_t> m_internal_sram;
+	memory_share_creator<uint32_t> m_apb;
+	memory_view m_remap_view;
+};
+
+void pixter_multimedia_state::apb_remap(uint32_t data)
+{
+	// User's Guide - 1.6 Memory Interface Architecture
+	if (data == 0) {
+		m_remap_view.select((m_apb[APB_PBC] & 0b0100) && (m_apb[APB_CS1OV] & 0b1) ? 0 : 3);
+	} else if (data < 3) {
+		m_remap_view.select(data);
+	} else {
+		logerror("Unexpected remap %d\n", data);
+	}
+}
+
+void pixter_multimedia_state::machine_start()
+{
+	if (m_cart->exists()) {
+		memory_region *cart_rom = m_cart->memregion("cart:rom");
```

Comment:
Shouldn’t this be `m_cart->memregion("rom")`?  When software list items are loaded, the data area tags are interpreted relative to the slot.  (The extra `cart:` part is only in the tag of redundant additional region that `common_load_rom` allocates.)

Reply:
I was missing the part of allocating with `m_cart->subtag("rom")` when loading manually, which makes this work, otherwise memregion wouldn't be able to find it. With the other part above it works for both cases now.